Factory/Config/Filter/ResizeFactory.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the `liip/LiipImagineBundle` project.
+ *
+ * (c) https://github.com/liip/LiipImagineBundle/graphs/contributors
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Liip\ImagineBundle\Factory\Config\Filter;
+
+use Liip\ImagineBundle\Config\Filter\Type\Resize;
+use Liip\ImagineBundle\Config\FilterInterface;
+use Liip\ImagineBundle\Factory\Config\FilterFactoryInterface;
+
+final class ResizeFactory implements FilterFactoryInterface
+{
+    const NAME = 'resize';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName(): string
+    {
+        return self::NAME;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(array $options): FilterInterface
+    {
+        $size = [];
+        if (isset($options['size']) && is_array($options['size'])) {
```

Comment:
this is silently ignoring non-array size options, which is a problem imho. we should do the `$size = $options['size'] ?? [];` . the constructor type declaration on $size will detect if the size option has been set to an invalid value.